airbyte-cdk/python/unit_tests/sources/test_abstract_source.py

```diff
@@ -1457,6 +1457,92 @@ def test_resumable_full_refresh_partial_failure(self, mocker):
+            {"records": [{"500": "500"}, {"600": "600"}], "next_page": {"page": 13}},
+            {"records": [{"700": "700"}, {"800": "800"}]},
+        ]
+        # When attempting to sync a slice, just output that slice as a record
+
+        # We've actually removed this filtering logic and will rely on the platform to dicate whether to pass state to the connector
+        # So in reality we can probably get rid of this test entirely
+        s1 = MockResumableFullRefreshStream(
+            [
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 10}}, responses[0]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 11}}, responses[1]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 12}}, responses[2]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 13}}, responses[3]),
+            ],
+            name="s1",
+        )
+
+        s2 = MockResumableFullRefreshStream(
+            [
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 10}}, responses[0]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 11}}, responses[1]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 12}}, responses[2]),
+                ({"stream_state": {"page": 10}, "sync_mode": SyncMode.full_refresh, "stream_slice": {"page": 13}}, responses[3]),
+            ],
+            name="s2",
+        )
+
+        mocker.patch.object(MockResumableFullRefreshStream, "get_json_schema", return_value={})
+
+        state = [
+            AirbyteStateMessage(
+                type=AirbyteStateType.STREAM,
+                stream=AirbyteStreamState(
+                    stream_descriptor=StreamDescriptor(name="s1"),
+                    stream_state=AirbyteStateBlob.parse_obj({"__ab_full_refresh_sync_complete": True}),
+                ),
+            ),
+            AirbyteStateMessage(
+                type=AirbyteStateType.STREAM,
+                stream=AirbyteStreamState(
+                    stream_descriptor=StreamDescriptor(name="s2"),
+                    stream_state=AirbyteStateBlob.parse_obj({"page": 10}),
+                ),
+            )
+        ]
+
+        src = MockSource(streams=[s1, s2])
+        catalog = ConfiguredAirbyteCatalog(
+            streams=[
+                _configured_stream(s1, SyncMode.full_refresh),
+                _configured_stream(s2, SyncMode.full_refresh),
+            ]
+        )
+
+        expected = _fix_emitted_at(
+            [
+                _as_stream_status("s1", AirbyteStreamStatus.STARTED),
+                _as_state("s1", {"__ab_full_refresh_sync_complete": True}),
```

Comment:
This is expected. Now that we perform the skip in the checkpoint reader, we'll still emit a single final state message w/ the existing sentinel value. And this keeps us consistent so every stream emits stream statuses and at least one state message